### ISSUE TITLE
[BUG] Document exists but search shows "no results" in Discover

### ISSUE THREAD

**Issue by antoniocascais** [opensearch-project/OpenSearch-Dashboards#2252]
**Describe the bug**

Sometimes, in certain searches, we get "no results" if we do the search via Opensearch dashboards but we do get results if we call Opensearch api directly. So we are 100% sure the document exists!

**To Reproduce**
1. Go to Opensearch dashboards and search for "field exists"
2. See "no results" message

**Expected behavior**
The expected behavior would be to actually see the document we are looking for.

**OpenSearch Version**
OpenSearch 1.3

**Dashboards Version**
OpenSearch Dashboards v 1.3.2

**Plugins**
* Query Workbench
* Reporting
* Alerting
* Anomaly Detection
* Observability
*Index Management

**Screenshots**
![image](https://user-images.githubusercontent.com/28305187/187963916-dfa14e0a-df60-4297-8769-f5289294e5f4.png)
![image](https://user-images.githubusercontent.com/28305187/187964363-eed0e5e6-a2c1-4b75-b4ad-d83902b51db9.png)

**Host/Environment (please complete the following information):**
 - OS: ArchLinux
 - Firefox v103
 - Opensearch service from AWS

**Additional context**

This problem also happens when we search for some keywords. There's no pattern, just some keywords show results and others don't.

Maybe it's a weird configuration problem somewhere, but have no idea what.

**Comment by joshuarrrr** [comment]
@antoniocascais From the screenshots, it appears your asking about "no results" in a Discover search. Have you verified that the time range matches the time range of the query your using in the console query? Another thing to check is whether the timestamp fields exist in your doc (necessary for Discover queries), that the index pattern definition is using the correct timestamp field, and that the timestamp field is formatted correctly.

**Comment by antoniocascais** [comment]
Hi @joshuarrrr 

Thank you for coming back to me. 

Our setup was 
```
fluent-bit -> aws kinesis firehose -> opensearch
```

We removed kinesis from the setup and now everything works fine. It's still weird to me why we couldn't search for certain logs (when the documents existed), but now it's fixed.

I can't check the timestamp now (because we had to recreate the cluster to remove kinesis), but I'm 99% sure that was not the problem. 

In some cases it happened that I was *seeing* the document in opensearch dashboards, but then when i tried to search for a UUID that was in that document i got no results back. If the problem would be the index pattern I wouldn't see anything in opensearch dashboards at all, right?

Anyway... problem solved on my side. Thanks :)
